tools/src/main/java/org/neo4j/tools/applytx/InteractiveApplyTransactions.java

```diff
@@ -0,0 +1,443 @@
+import org.neo4j.kernel.api.exceptions.TransactionFailureException;
+import org.neo4j.kernel.api.labelscan.LabelScanStore;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.impl.api.BatchingTransactionRepresentationStoreApplier;
+import org.neo4j.kernel.impl.api.LegacyIndexApplierLookup;
+import org.neo4j.kernel.impl.api.TransactionRepresentationCommitProcess;
+import org.neo4j.kernel.impl.api.index.IndexUpdatesValidator;
+import org.neo4j.kernel.impl.api.index.IndexingService;
+import org.neo4j.kernel.impl.core.CacheAccessBackDoor;
+import org.neo4j.kernel.impl.index.IndexConfigStore;
+import org.neo4j.kernel.impl.locking.LockGroup;
+import org.neo4j.kernel.impl.locking.LockService;
+import org.neo4j.kernel.impl.pagecache.StandalonePageCacheFactory;
+import org.neo4j.kernel.impl.store.StoreAccess;
+import org.neo4j.kernel.impl.store.record.NodeRecord;
+import org.neo4j.kernel.impl.store.record.PropertyRecord;
+import org.neo4j.kernel.impl.store.record.Record;
+import org.neo4j.kernel.impl.store.record.RelationshipGroupRecord;
+import org.neo4j.kernel.impl.store.record.RelationshipRecord;
+import org.neo4j.kernel.impl.transaction.CommittedTransactionRepresentation;
+import org.neo4j.kernel.impl.transaction.log.IOCursor;
+import org.neo4j.kernel.impl.transaction.log.LogicalTransactionStore;
+import org.neo4j.kernel.impl.transaction.log.ReadOnlyTransactionStore;
+import org.neo4j.kernel.impl.transaction.log.TransactionIdStore;
+import org.neo4j.kernel.impl.transaction.state.NeoStoreProvider;
+import org.neo4j.kernel.impl.transaction.tracing.CommitEvent;
+import org.neo4j.kernel.impl.util.IdOrderingQueue;
+import org.neo4j.kernel.impl.util.StringLogger;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.kernel.monitoring.Monitors;
+import org.neo4j.tooling.GlobalGraphOperations;
+import org.neo4j.tools.CommandReactor;
+import org.neo4j.tools.CommandReactor.Action;
+
+import static org.neo4j.helpers.progress.ProgressMonitorFactory.textual;
+import static org.neo4j.kernel.impl.api.TransactionApplicationMode.RECOVERY;
+
+/**
+ * Interactive apply transactions from transaction logs onto a new store, effectively rebuilding a database
+ * from transaction logs. Transaction can be applied interactive, i.e. applied up to any transaction id
+ * and consistency checked at any point. Also there are other utilities, such as printing record structure
+ * at any point as well.
+ *
+ * Running this tool will eventually go into a mode where it's awaiting user input, so typed commands
+ * will control what happens. For example:
+ * <ul>
+ * <li>apply 1234 : applies transactions from current point and up to an including 1234</li>
+ * <li>apply next : applies next transaction</li>
+ * <li>cc : runs consistency checker on the built database at this point</li>
+ * <li>propchain 456 : prints property record chain for node 456</li>
+ * <li>relchain 789 : prints relationship record chain for relationship 789</li>
```

Comment:
the relationship chain for a relationship? how do I print the relationship chain for a node?